pkgs/applications/video/vivictpp/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv, fetchFromGitHub
+, meson, cmake, ninja, pkg-config
+, python3, git
+, SDL2, SDL2_ttf
+, freetype
+, ffmpeg
+, cacert }:
+
+let
+  withSubprojects = stdenv.mkDerivation {
+    name = "sources-with-subprojects";
+
+    src = fetchFromGitHub {
+      owner = "vivictorg";
+      repo = "vivictpp";
+      rev = "f682bec15cf69faa3c37b8fc5e27408b89da2281";
+      hash = "sha256-hb5LCPsoda27oCJrhZaWq3ov+SmMLdECXvTXfrXFJgo=";
+    };
+
+    nativeBuildInputs = [
+      meson
+      cacert
+      git
+    ];
+
+    buildCommand = ''
+      cp -r --no-preserve=mode $src $out
+      cd $out
+
+      meson subprojects download
+      find subprojects -type d -name .git -prune -execdir rm -r {} +
+    '';
+
+    outputHashMode = "recursive";
+    outputHash = "sha256-J7B9l03r+D09a1mj6aChZfLHgYfNOJUHIpPr5LORKAE=";
+  };
+in stdenv.mkDerivation rec {
+  pname = "vivictpp";
+  version = "0.2.5";
+
+  src = withSubprojects;
+
+  nativeBuildInputs = [
+    meson
+    cmake
+    ninja
+    pkg-config
+
+    python3
+    git
+  ];
+
+  buildInputs = [
+    SDL2
+    SDL2_ttf
+    freetype
+    ffmpeg
+  ];
+
+  preConfigure = ''
+    patchShebangs .
+  '';
+
+  meta = with lib; {
+    description = "An easy to use tool for subjective comparison of the visual quality of different encodings of the same video source";
+    homepage = "https://github.com/vivictorg/vivictpp";
+    license = licenses.gpl2Plus;
+    platforms = platforms.unix;
```

Comment:
This is a guess.

The [project's README](https://github.com/vivictorg/vivictpp#other-platforms) says

> Vivict++ is intended to be cross platform, so building on Windows or macOS should be possible. Testing and help would be appreciated.

I have no easy way of testing whether it builds/runs on darwin, or non-linux unix variants.